rust/src/http2/detect.rs

```diff
@@ -0,0 +1,737 @@
+) -> std::os::raw::c_int {
+    for i in 0..hd.blocks.len() {
+        match ctx.mode {
+            parser::DetectUintMode::DetectUintModeEqual => {
+                if hd.blocks[i].sizeupdate == ctx.value
+                    && hd.blocks[i].error
+                        == parser::HTTP2HeaderDecodeStatus::HTTP2HeaderDecodeSizeUpdate
+                {
+                    return 1;
+                }
+            }
+            parser::DetectUintMode::DetectUintModeLt => {
+                if hd.blocks[i].sizeupdate <= ctx.value
+                    && hd.blocks[i].error
+                        == parser::HTTP2HeaderDecodeStatus::HTTP2HeaderDecodeSizeUpdate
+                {
+                    return 1;
+                }
+            }
+            parser::DetectUintMode::DetectUintModeGt => {
+                if hd.blocks[i].sizeupdate >= ctx.value
+                    && hd.blocks[i].error
+                        == parser::HTTP2HeaderDecodeStatus::HTTP2HeaderDecodeSizeUpdate
+                {
+                    return 1;
+                }
+            }
+            parser::DetectUintMode::DetectUintModeRange => {
+                if hd.blocks[i].sizeupdate <= ctx.value
+                    && hd.blocks[i].sizeupdate >= ctx.valrange
+                    && hd.blocks[i].error
+                        == parser::HTTP2HeaderDecodeStatus::HTTP2HeaderDecodeSizeUpdate
+                {
+                    return 1;
+                }
+            }
+        }
+    }
+    return 0;
+}
+
+fn http2_detect_sizeupdatectx_match(
+    ctx: &mut parser::DetectU64Data, tx: &mut HTTP2Transaction, direction: u8,
+) -> std::os::raw::c_int {
+    if direction & STREAM_TOSERVER != 0 {
+        for i in 0..tx.frames_ts.len() {
+            match &tx.frames_ts[i].data {
+                HTTP2FrameTypeData::HEADERS(hd) => {
+                    if http2_detect_sizeupdate_match(&hd, ctx) != 0 {
+                        return 1;
+                    }
+                }
+                _ => {}
+            }
+        }
+    } else {
+        for i in 0..tx.frames_tc.len() {
+            match &tx.frames_tc[i].data {
+                HTTP2FrameTypeData::HEADERS(hd) => {
+                    if http2_detect_sizeupdate_match(&hd, ctx) != 0 {
+                        return 1;
+                    }
+                }
+                _ => {}
+            }
+        }
+    }
+    return 0;
+}
+
+#[no_mangle]
+pub extern "C" fn rs_http2_detect_sizeupdatectx_match(
+    ctx: *const std::os::raw::c_void, tx: *mut std::os::raw::c_void, direction: u8,
+) -> std::os::raw::c_int {
+    let ctx = cast_pointer!(ctx, parser::DetectU64Data);
+    let tx = cast_pointer!(tx, HTTP2Transaction);
+    return http2_detect_sizeupdatectx_match(ctx, tx, direction);
+}
+
+//TODOask better syntax between rs_http2_tx_get_header_name in argument
+// and rs_http2_detect_sizeupdatectx_match explicitly casting
+#[no_mangle]
```

Comment:
What is the best syntax ? 
Argument `tx: *mut std::os::raw::c_void` and then`let tx = cast_pointer!(tx, HTTP2Transaction);`
Or argument `tx: &mut HTTP2Transaction`